dali/kernels/slice/slice_flip_normalize_permute_cpu.h

```diff
@@ -0,0 +1,272 @@
+                                   const std::array<int64_t, Dims> &in_strides,
+                                   const std::array<int64_t, Dims> &out_strides,
+                                   const TensorShape<static_cast<int>(Dims)> &out_shape,
+                                   const TensorShape<static_cast<int>(Dims)> &padded_out_shape,
+                                   const float *mean, const float *inv_stddev,
+                                   size_t normalization_dim,
+                                   const int64_t *permuted_dims,
+                                   std::integral_constant<size_t, 1>) {
+  constexpr auto d = Dims - 1;
+  for (int i = 0; i < out_shape[d]; i++) {
+    size_t norm_idx = normalization_dim == d ? i : 0;
+    Policy::Fill(*output, *input, mean + norm_idx, inv_stddev + norm_idx);
+    input += in_strides[d];
+    output += out_strides[d];
+  }
+
+  for (int i = out_shape[d]; i < padded_out_shape[d]; i++) {
+    size_t norm_idx = normalization_dim == d ? i : 0;
+    ZeroPadPolicy::Fill(*output, *input, mean + norm_idx, inv_stddev + norm_idx);
+    input += in_strides[d];
+    output += out_strides[d];
+  }
+}
+
+template <typename Policy, typename OutputType, typename InputType, size_t Dims, size_t DimsLeft>
+void SliceFlipNormalizePermuteImpl(OutputType *output, const InputType *input,
+                                   const std::array<int64_t, Dims> &in_strides,
+                                   const std::array<int64_t, Dims> &out_strides,
+                                   const TensorShape<static_cast<int>(Dims)> &out_shape,
+                                   const TensorShape<static_cast<int>(Dims)> &padded_out_shape,
+                                   const float *mean, const float *inv_stddev,
+                                   size_t normalization_dim,
+                                   const int64_t *permuted_dims,
+                                   std::integral_constant<size_t, DimsLeft>) {
+  constexpr auto d = Dims - DimsLeft;
+
+  for (int i = 0; i < out_shape[d]; i++) {
+    size_t norm_idx = normalization_dim == d ? i : 0;
+    SliceFlipNormalizePermuteImpl<Policy>(output, input, in_strides, out_strides, out_shape,
+                                          padded_out_shape, mean + norm_idx, inv_stddev + norm_idx,
+                                          normalization_dim, permuted_dims,
+                                          std::integral_constant<size_t, DimsLeft - 1>());
+    input += in_strides[d];
+    output += out_strides[d];
+  }
+
+  for (int i = out_shape[d]; i < padded_out_shape[d]; i++) {
+    size_t norm_idx = normalization_dim == d ? i : 0;
+    SliceFlipNormalizePermuteImpl<ZeroPadPolicy>(
+        output, input, in_strides, out_strides, out_shape, padded_out_shape, mean + norm_idx,
+        inv_stddev + norm_idx, normalization_dim, permuted_dims,
+        std::integral_constant<size_t, DimsLeft - 1>());
+    input += in_strides[d];
```

Comment:
No need. Input is already out of bounds at this point.

Reply:
done